### DISCUSSION TITLE
Duplicate conversations and contacts Twilio inbox with Whatsapp

### DISCUSSION THREAD

**Discussion by YanLipu** [chatwoot/chatwoot#9824] [Q&A]
TL/DR: I'm facing almost the same issue related [here](https://github.com/chatwoot/chatwoot/issues/5840). The difference is that I have a Twilio Inbox integration instead of a WhatsApp Cloud.

Describe the bug

Hello. Here in Brazil, we had a change on mobile numbers a few years ago, they've added an extra leading digit to account the total numbers growth over the last years. The general change was to add a leading 9 over all current numbers. Most WA users have their accounts registered to the old number, without the leading digit. When we start a conversation with Chatwoot, we add the correct number, with the leading digit but when the user responds, Chatwoot creates both a new contact and a new conversation ticket with the registered, old number, without the leading number. This is a major problem, because it breaks the ticket flow and duplicates contacts.

To Reproduce

Create a contact with the "new", correct number, then start a conversation, wait for them to respond, check conversations and contacts.

Expected behavior

Contacts and conversations should not duplicate.

Environment

I'm currently using self hosted Chatwoot with Twilio inbox.

Can anyone help me?

**Comment by muhsin-k** [comment]
Will fix via https://github.com/chatwoot/chatwoot/pull/12655
